Docs/sphinx_documentation/source/BuildingAMReX.rst

```diff
@@ -28,37 +28,40 @@ list of important variables.
-   +------------+-------------------------------------+--------------------+
-   | WARN_ALL   | TRUE or FALSE                       | TRUE for DEBUG     |
-   |            |                                     | FALSE otherwise    |
-   +------------+-------------------------------------+--------------------+
+   +-----------------+-------------------------------------+--------------------+
+   | Variable        | Value                               | Default            |
+   +=================+=====================================+====================+
+   | AMREX_HOME      | Path to amrex                       | environment        |
+   +-----------------+-------------------------------------+--------------------+
+   | COMP            | gnu, cray, ibm, intel, llvm, or pgi | none               |
+   +-----------------+-------------------------------------+--------------------+
+   | CXXSTD          | C++ standard (``c++11``, ``c++14``, | compiler default,  |
+   |                 | ``c++17``, ``c++20``)               | at least ``c++11`` |
+   +-----------------+-------------------------------------+--------------------+
+   | DEBUG           | TRUE or FALSE                       | FALSE              |
+   +-----------------+-------------------------------------+--------------------+
+   | DIM             | 1 or 2 or 3                         | 3                  |
+   +-----------------+-------------------------------------+--------------------+
+   | PRECISION       | DOUBLE or FLOAT                     | DOUBLE             |
+   +-----------------+-------------------------------------+--------------------+
+   | USE_MPI         | TRUE or FALSE                       | FALSE              |
+   +-----------------+-------------------------------------+--------------------+
+   | USE_OMP         | TRUE or FALSE                       | FALSE              |
+   +-----------------+-------------------------------------+--------------------+
+   | USE_CUDA        | TRUE or FALSE                       | FALSE              |
+   +-----------------+-------------------------------------+--------------------+
+   | USE_HIP         | TRUE or FALSE                       | FALSE              |
+   +-----------------+-------------------------------------+--------------------+
+   | USE_DPCPP       | TRUE or FALSE                       | FALSE              |
+   +-----------------+-------------------------------------+--------------------+
+   | USE_RPATH       | TRUE or FALSE                       | FALSE              |
+   +-----------------+-------------------------------------+--------------------+
+   | WARN_ALL        | TRUE or FALSE                       | TRUE for DEBUG     |
+   |                 |                                     | FALSE otherwise    |
+   +-----------------+-------------------------------------+--------------------+
+   | AMREX_CUDA_ARCH | CUDA arch such as 70                | 70 if not set      |
```

Comment:
@WeiqunZhang looking at this PR again, I actually meant to also introduce an **environment** variable `AMREX_CUDA_ARCH` that sets the default for this GNUmake option.

Maybe I don't get the implementation, but at the moment it looks like this PR added a GNUmake option that is defaulted to `70` but cannot be defaulted with an env var.

I like to introduce env-var control in WarpX, because it simplifies modules and `warpx.profile` files for users with sensible defaults for a system.

Reply:
If the environment defines `AMREX_CUDA_ARCH`.  `ifdef AMREX_CUDA_ARCH` will be true.  So it looks OK to me.

Reply:
Perfect, I was confused if that is picked up. Thanks! :)